CMakeLists.txt

```diff
@@ -29,6 +29,8 @@ include(DispatchAppleOptions)
 option(ENABLE_DISPATCH_INIT_CONSTRUCTOR "enable libdispatch_init as a constructor" ON)
 set(USE_LIBDISPATCH_INIT_CONSTRUCTOR ${ENABLE_DISPATCH_INIT_CONSTRUCTOR})
 
+option(BUILD_SHARED_LIBS "build shared libraries" ON)
```

Comment:
Do we want to default to the shared libraries first?  We can, I went with the defaults of auto tools, which was static by default.